content/docs/guides/branching-github-actions.md

```diff
@@ -286,6 +287,137 @@ outputs:
 - `host_with_pooler`: The branch host with pooling after the reset.
 - `password`: The password for connecting to the branch database after the reset.
 
+## Schema Diff action
+
+This action performs a database schema diff on specified Neon branches for each pull request and writes a **Neon Schema Diff summary** comment to the pull request in GitHub highlighting the schema differences.
+
+It supports workflows where schema changes are made on a development branch, and pull requests are created for review before merging the changes back into the main branch. By including the schema diff as a comment in the pull request, reviewers can easily assess the changes directly within the pull request.
+
+<Admonition type="tip" title="GitHub Actions Marketplace">
+You can find this action on the **GitHub Actions Marketplace**: [Neon Schema Diff GitHub Action](https://github.com/marketplace/actions/neon-schema-diff-github-action).
+</Admonition>
+
+### Prerequisites
+
+Using the action requires adding a Neon API key to your GitHub Secrets. For information about obtaining an API key, see [Create an API key](/docs/manage/api-keys#create-an-api-key).
+
+1. In your GitHub repository, go to **Project settings** and locate **Secrets** at the bottom of the left sidebar.
+2. Click **Actions** > **New Repository Secret**.
+3. Name the secret `NEON_API_KEY` and paste your API key in the **Secret** field
+4. Click **Add Secret**.
+
+Alternatively, the **Neon GitHub Integration** can perform the API key setup for you. See [Neon GitHub integration](/docs/guides/neon-github-integration).
+
+### Example
+
+The following example performs a schema diff on a database named `mydatabase` between the `compare_branch` and the `base_branch` branch.
+
+```yaml
+steps:
+  - uses: neondatabase/schema-diff-action@v1
+    with:
+      project_id: ${{ vars.NEON_PROJECT_ID }}
+      compare_branch: preview/pr-${{ github.event.number }}-${{ needs.setup.outputs.branch }}
+      base_branch: main
+      api_key: ${{ secrets.NEON_API_KEY }}
+      database: mydatabase
+      username: myrole
+```
+
+Here's an example workflow that incorporates the action:
+
+```yaml
+name: Schema Diff for Pull Requests
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+
+jobs:
+  schema_diff:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Schema Diff
+        uses: neondatabase/schema-diff-action@v1
+        with:
+          project_id: ${{ vars.NEON_PROJECT_ID }}
+          compare_branch: preview/pr-${{ github.event.number }}
+          base_branch: main
+          api_key: ${{ secrets.NEON_API_KEY }}
+          database: mydatabase
+          username: myrole
```

Comment:
Update workflow to include the permissions section

Reply:
`permissions: write-all`

Reply:
Added. Also mentioned the less permissive alternative.